backend/app/services/auth.py

```
@@ -0,0 +1,188 @@
+            import base58
+            
+            # Decode the public key and signature
+            pubkey_bytes = base58.b58decode(wallet_address)
+            signature_bytes = base58.b58decode(signature)
+            message_bytes = message.encode('utf-8')
+            
+            # Verify the signature
+            verify_key = VerifyKey(pubkey_bytes)
+            verify_key.verify(message_bytes, signature_bytes)
+            verified = True
+        except (BadSignature, Exception) as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)